### PR TITLE
Realtime chat

### PR DESCRIPTION
**Pull request by FedericoLeiva12**
I have created a `chat` page so I can use the home page for the sign in feature.

If you are in the `sign in` page and you're already logged, it will automatically redirect you to the `chat` page.

If you are in the `chat` page but you are not logged, it will automatically redirect you to the `sign in` page.

I'm just using the base components from shadcn but I don't know if there are designs done for this. If you send me a design I can work on it.

https://github.com/The-Creative-Programming-Group/Convex-Chat-App/assets/49959270/6f32f810-40f4-478d-a5d2-454292322034

### CODE REVIEW

**Comment by FleetAdmiralJakob** [comment]
@FedericoLeiva12 May I ask a question: Why did you change the _generated files? To add missing types? Because I thought they are generated and will be overwritten if I type the dev command.

**Comment by FleetAdmiralJakob** [comment]
@FedericoLeiva12 Can you fix these Typescript errors?

**Comment by FedericoLeiva12** [comment]
> @FedericoLeiva12 May I ask a question: Why did you change the _generated files? To add missing types? Because I thought they are generated and will be overwritten if I type the dev command.

The _generated files changed automatically when I ran the dev command. I wasn't a manual change.

**Comment by FedericoLeiva12** [comment]
> @FedericoLeiva12 Can you fix these Typescript errors? 

Should we exclude the ui folder from typescript check? These files was added by shadcn.

**Comment by FleetAdmiralJakob** [comment]
> > @FedericoLeiva12 May I ask a question: Why did you change the _generated files? To add missing types? Because I thought they are generated and will be overwritten if I type the dev command.
> 
> The _generated files changed automatically when I ran the dev command. I wasn't a manual change.

Okay, cool

**Comment by FleetAdmiralJakob** [comment]
> > @FedericoLeiva12 Can you fix these Typescript errors? 
> 
> Should we exclude the ui folder from typescript check? These files was added by shadcn.

No, I think we should just edit the UI components

File: src/ui/ui/input.tsx [hunk 1]
```diff
@@ -0,0 +1,25 @@
+import * as React from "react";
+
+import { cn } from "~/lib/utils";
+
+export interface InputProps
+  extends React.InputHTMLAttributes<HTMLInputElement> {}
```
Comment (FleetAdmiralJakob) [inline src/ui/ui/input.tsx]: I think the problem is here that typescript does not allow these empty brackets

File: src/ui/ui/textarea.tsx [hunk 2]
```diff
@@ -0,0 +1,24 @@
+import * as React from "react";
+
+import { cn } from "~/lib/utils";
+
+export interface TextareaProps
+  extends React.TextareaHTMLAttributes<HTMLTextAreaElement> {}
```
Comment (FleetAdmiralJakob) [inline src/ui/ui/textarea.tsx]: I think the problem is here that typescript does not allow these empty brackets

**Review by FleetAdmiralJakob — CHANGES_REQUESTED** [review]
I will need to look to the deploy before I will merge it

**Review by FleetAdmiralJakob — APPROVED** [review]

**Pull request merged**